cmd/dep/init.go

```diff
@@ -180,6 +180,17 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	}
 	l = dep.LockFromInterface(soln)
 
+	// Find any solved dependencies that are unconstrained and remove them from
+	// the manifest.
+	for pr, d := range m.Dependencies {
+		if d.Constraint == nil {
```

Comment:
Also need to check if `d.Source == ""` - we only consider a project empty/unconstrained if it has neither a version constraint, nor a special source location.